stdlib/bytearray.mini

```diff
@@ -129,165 +126,139 @@ public func bytearray_size(ba: ByteArray) -> uint {
-        }
-        return ret;
     }
+    offset = offset + ba.sliceOffset;
+    let res = getbuffer64(ba.contents, offset);
+    return res;
 }
 
 // bytearray_get256 reads a chunk of 32 bytes from a ByteArray
 public func bytearray_get256(ba: ByteArray, offset: uint) -> uint {
     if (offset >= ba.size) {
         return 0;
     }
-    let bytesToTrim = 0;
     if (offset+32 > ba.size) {
-        bytesToTrim = offset+32 - ba.size;
-    }
-    offset = offset + ba.sliceOffset;
-    let lowWordIndex = offset / 32;
-    let alignment = offset % 32;
-    if (alignment == 0) {
-        let ret = expandingIntArray_get(ba.contents, lowWordIndex);
-        if (bytesToTrim > 0) {
-            ret = ret & ~(asm(8*bytesToTrim, 1) uint { shl } - 1);
-        }
-        return ret;
-    } else {
-        let (lowWord, hiWord) = expandingIntArray_getConsecutive(ba.contents, lowWordIndex);
-        let shiftFactor = asm(8*alignment, 1) uint { shl };
-        let invShiftFactor = asm(8*(32-alignment), 1) uint { shl };
-        let ret = (hiWord/invShiftFactor) | (lowWord*shiftFactor);
-        if (bytesToTrim > 0) {
-            ret = ret & ~(asm(8*bytesToTrim, 1) uint { shl } - 1);
+        let bytesToRead = ba.size-offset;
+        offset = offset + ba.sliceOffset;
+        if (offset + bytesToRead >= 32) {
+            let mask = asm(8*bytesToRead, 1) uint { shl } - 1;
+            return asm(8*(32-bytesToRead), mask & getbuffer256(ba.contents, offset+bytesToRead-32)) uint { shl };
+        } else {
+            return getbuffer256(ba.contents, offset);
         }
-        return ret;
     }
+    offset = offset + ba.sliceOffset;
+    let res = getbuffer256(ba.contents, offset);
+    return res;
+}
+
+func make(cont: ByteArray) -> ByteArray {
+    return struct {
+        size: cont.size + cont.sliceOffset + 32,
+        sliceOffset: 0,
+        contents: cont.contents,
+        copy: true,
+    };
+}
+
+public func bytearray_wrap(cont: buffer, size: uint) -> ByteArray {
+    return struct {
+        size: size,
+        sliceOffset: 0,
+        contents: cont,
+        copy: false,
+    };
 }
 
 // bytearray_setbyte writes one byte to a ByteArray, returns the resulting ByteArray
-public func bytearray_setByte(ba: ByteArray, offset: uint, val: uint) -> ByteArray {
-    if (offset >= ba.size) {
-        if (offset >= 0x100000000000000000000000000000000) {
-            return ba;    // ignore out-of-bounds write
-        }
-        ba = ba with { size: offset+1 };
+public func bytearray_setByte(ba: ByteArray, offset: uint, value: uint) -> ByteArray {
+    if (offset >= const::MaxByteArraySize) {
+        return ba;    // ignore out-of-bounds write
     }
-    offset = offset + ba.sliceOffset;
-    let (newContents, _) = expandingIntArray_op(
-        ba.contents, 
-        offset/32, 
-        struct {
-            f: unsafecast<OpClosureFunc>(basb_closure_func),
-            val: (31-(offset%32), val,),
+    if (offset >= const::MaxByteArraySize) {
```

Comment:
This conditional repeats the first and should be corrected

Reply:
Fixed.